### PR TITLE
feat: add module resume upload and parse

### PR DESCRIPTION
**Pull request by Stanford997**
## Context

dd module resume upload and parse

## Changes

dd module resume upload and parse

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: code/modules/parser.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+"""
+This module is responsible for converting PDF resumes into string text format.
+"""
+
+from PyPDF2 import PdfReader
+
```
Comment (adamma1024) [inline code/modules/parser.py:6]: Redundant line?

File: code/modules/parser.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+def convert_pdf_to_text(file_path):
+    try:
+        reader = PdfReader(file_path)
+        text = ''
+        for page in reader.pages:
+            text += page.extract_text()
+        return text
+    except Exception as e:
+        return f"Error processing PDF: {str(e)}"
+
```
Comment (adamma1024) [inline code/modules/parser.py:17]: Same

File: code/modules/upload.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
+"""
+This module is responsible for upload user's resume and save it in MongoDB
+"""
+from flask import jsonify
+from modules.parser import parse_resume
+
+ALLOWED_EXTENSIONS = {'pdf'}
+
+
```
Comment (adamma1024) [inline code/modules/upload.py:9]: The py rule asks there should be two empty lines here?

File: code/modules/upload.py [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+# PDF format check
+def allowed_file(filename):
+    return '.' in filename and filename.rsplit('.', 1)[1].lower() in ALLOWED_EXTENSIONS
+
+
+def upload_parse_resume(request, resume_collection):
+    if 'file' not in request.files:
+        return jsonify({"error": "No file part in the request"}), 400
+
+    file = request.files['file']
```
Comment (adamma1024) [inline code/modules/upload.py:19]: So the interface is like:
```py
user_id: string
file: blob
```
correct?

File: code/modules/upload.py (hunk 3 above)
Comment (Stanford997) [reply code/modules/upload.py:9]: Yes. You are right. PEP8 requires 2 blank lines before each def in a file.

File: code/modules/parser.py (hunk 2 above)
Comment (Stanford997) [reply code/modules/parser.py:17]: PEP8 requires 2 blank lines before each def in a file.

File: code/modules/parser.py (hunk 1 above)
Comment (Stanford997) [reply code/modules/parser.py:6]: PEP8 requires 2 blank lines before each def in a file.

Comment (Stanford997) [reply code/modules/parser.py:6]: According to PEP 8 (Python's style guide), it's a good practice to have one blank line between the module-level docstring and import statements.
Also for brevity, this line can be removed.

File: code/modules/upload.py (hunk 4 above)
Comment (Stanford997) [reply code/modules/upload.py:19]: correct~~~

File: code/modules/upload.py (hunk 3 above)
Comment (adamma1024) [reply code/modules/upload.py:9]: Sure.

File: code/Readme.md [hunk 5]
```diff
@@ -1 +1,3 @@
 This folder contains all source code and test code.
+
+Test for push.
```
Comment (adamma1024) [inline code/Readme.md:3]: Don't forget to remove this line.
And btw, please check our [Conventional Commit rules](https://www.conventionalcommits.org/en/v1.0.0/) to refine the commit message. @andyasdd1

**Review by adamma1024 — APPROVED** [review]

**Pull request merged**